components/esp_modem/test/target_ota/components/manual_ota/transport_batch_tls.cpp

```diff
@@ -0,0 +1,232 @@
+    std::vector<char> buf;
+    size_t read_len;
+    size_t offset;
+};
+
+esp_transport_handle_t esp_transport_tls_init(esp_transport_handle_t parent)
+{
+    esp_transport_handle_t ssl = esp_transport_init();
+    auto *tls = new TlsTransport(parent);
+    esp_transport_set_context_data(ssl, tls);
+    TlsTransport::set_func(ssl);
+    return ssl;
+}
+
+int TlsTransport::send(const unsigned char *buf, size_t len)
+{
+    int ret = esp_transport_write(transport_, reinterpret_cast<const char *>(buf), len, 0);
+    ESP_LOGD(TAG, "writing(len=%d) ret=%d", len, ret);
+    return ret;
+}
+
+int TlsTransport::recv(unsigned char *buffer, size_t len)
+{
+    ESP_LOGD(TAG, "recv(len=%d)", len);
+    if (read_len != 0) {
+
+        if (read_len > len) {
+            memcpy((char *)buffer, buf.data() + offset, len);
+            read_len -= len;
+            offset += len;
+            ESP_LOGD(TAG, "read %d from batch read_len = %d", len, read_len);
+            return len;
+        } else {
+            int remaining = len = read_len;
+            if (remaining > 0) {
+                memcpy((char *)buffer, buf.data() + offset, remaining);
+                read_len = 0;
+                offset = 0;
+                return remaining;
+
+            }
+            read_len = 0;
+            offset = 0;
+            return ERR_TCP_TRANSPORT_CONNECTION_CLOSED_BY_FIN;
+        }
+    }
+    int ret = esp_transport_read(transport_, reinterpret_cast<char *>(buffer), len, last_timeout);
+
+    if (ret == ERR_TCP_TRANSPORT_CONNECTION_TIMEOUT) {
+        return MBEDTLS_ERR_SSL_WANT_READ;
+    }
+    return ret == ERR_TCP_TRANSPORT_CONNECTION_CLOSED_BY_FIN ? 0 : ret;
+}
+
+bool TlsTransport::set_func(esp_transport_handle_t tls_transport)
+{
+    return esp_transport_set_func(tls_transport, TlsTransport::transport::connect, TlsTransport::transport::read, TlsTransport::transport::write, TlsTransport::transport::close, TlsTransport::transport::poll_read, TlsTransport::transport::poll_write, TlsTransport::transport::destroy) == ESP_OK;
+}
+
+int TlsTransport::connect(const char *host, int port, int timeout_ms)
+{
+    return esp_transport_connect(transport_, host, port, timeout_ms);
+}
+
+void TlsTransport::delay()
+{
+    vTaskDelay(pdMS_TO_TICKS(500));
+}
+
+int TlsTransport::transport::connect(esp_transport_handle_t t, const char *host, int port, int timeout_ms)
+{
+    auto tls = static_cast<TlsTransport *>(esp_transport_get_context_data(t));
+    tls->init(is_server{false}, do_verify{false});
+
+    ESP_LOGD(TAG, "TLS-connect");
+    auto ret = tls->connect(host, port, timeout_ms);
+    if (ret < 0) {
+        ESP_LOGI(TAG, "Failed to connect to transport");
+        return ret;
+    }
+    if (tls->is_session_loaded()) {
+        tls->set_session();
+    }
+    ESP_LOGI(TAG, "Before handshake");
+    ret = tls->handshake();
+    if (ret < 0) {
+        ESP_LOGI(TAG, "Failed to handshake");
+        return ret;
+    }
+    tls->get_session();
```

Comment:
Does `tls->get_session()` return a status?
If yes, should that be returned instead of hardcoded `0`?

Reply:
good point, needs an error check